src/agentlab/analyze/agent_xray.py

```diff
@@ -648,7 +648,10 @@ def get_episode_info(info: Info):
         env_args = info.exp_result.exp_args.env_args
         steps_info = info.exp_result.steps_info
         step_info = steps_info[info.step]
-        goal = step_info.obs["goal"]
+        try:
+            goal = step_info.obs["goal"]
+        except:
```

Comment:
avoid generic except, this can lead to silent error. I think it's `except KeyError` or something like that.